added VNet peering info (#1053)
astronomer/docs: astro/install-azure.md

For more information about the resources required to run Astro on Azure, see [Azure Resource Reference](resource-reference-azure.md).

### VNet peering prerequisites (Optional)

> If any Azure resources are on a private network, you can choose between two options:

Reviewer: ```suggestion
If you want to access Azure resources on a private network, you have the following two options:
```

> 1. Create a VNet Peering connection between Astronomer's VNet and the VNets for your broader network. 

Reviewer: ```suggestion
- Create a VNet peering connection between the Astronomer VNet and your network VNets. 
```

Reviewer 2: It should say Astro VNet - not Astronomer VNet?  Because the VNet is for the product, not the company?

> 2. [Virtual Network Integration](https://docs.microsoft.com/en-us/azure/api-management/virtual-network-concepts?tabs=stv2)

Reviewer: ```suggestion
- [Virtual Network Integration](https://docs.microsoft.com/en-us/azure/api-management/virtual-network-concepts?tabs=stv2)
```

Reviewer 2: We shouldn't support Virtual Network Integration - because Virtual Network Integration means a non-Astro service is running "being injected into" (Azure's words, not mine) the VNet.  That isn't the Astro model. We don't want any of their stuff in our VNet.

PS: This means we need to add documentation for Private Endpoints, similar to how we did it for [AWS](https://docs.astronomer.io/astro/connect-external-services#aws-privatelink)  - I think that should probably be a different issue/PR than this one.

> If you want to continue with the option #1, please open a ticket within our [Astronomer Support](https://support.astronomer.io/) with following details and support will proceed on configuring the VNet peering:

Reviewer: ```suggestion
To create a VNet peering connection, submit a  [support request](astro-support.md) with the following information:
```

> - Azure TenantID of the target VNet

Reviewer: ```suggestion
- Azure TenantID of the target VNet.
```

> - Subscription ID of the target VNet

Reviewer: ```suggestion
- Subscription ID of the target VNet.
```

> - Region of the target VNet

Reviewer: ```suggestion
- Region of the target VNet.
```

> - VNet ID of the target VNet

Reviewer: ```suggestion
- VNet ID of the target VNet.
```

> - A CIDR block (RFC 1918 IP Space) no smaller than a `/19` range. You must ensure it does not overlap with the Azure VNet(s) that you will be peering with later. The default CIDR range is `172.20.0.0/19`.

Reviewer: ```suggestion
- A CIDR block (RFC 1918 IP Space) no smaller than a `/19` range. The CIDR block shouldn't overlap with the Azure VNet(s) that you're peering with. The default CIDR range is `172.20.0.0/19`.
```

> - Resource group name (you can reference the [Management of Azure Resource Groups Docs here.](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal). [Azure Portal login](https://portal.azure.com/)

Reviewer: ```suggestion
- The resource group name. See [Manage Azure resource groups by using the Azure portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal) or  [log in to the Azure Portal](https://portal.azure.com/)
```

Reviewer 2: Since Astro creates the resource group (its the name of the cluster), we know the resource group name (or can divine it).  I would rather not ask them to log into this subscription and take a peek (I know they can and will do it, but its just not our model so best to avoid asking them to do it)
